keel-core/src/main/kotlin/com/netflix/spinnaker/keel/constraints/ConstraintEvaluator.kt

```diff
@@ -56,7 +57,8 @@ inline fun <reified T : Constraint> SupportedConstraintType(name: String): Suppo
   SupportedConstraintType(name, T::class.java)
 
 abstract class StatefulConstraintEvaluator<T : Constraint>(
-  protected val repository: KeelRepository
+  protected val repository: KeelRepository,
+  protected open val authorizationSupport: AuthorizationSupport
```

Comment:
One downside of this approach is having to inject `AuthorizationSupport` everywhere you want to inject those headers. 😞